From 913e6eb1081b5e3bf494ba9496f6b17addf9b15e Mon Sep 17 00:00:00 2001
From: Martino Facchin <m.facchin@arduino.cc>
Date: Thu, 24 Jul 2025 11:52:58 +0200
Subject: [PATCH 09/17] gpu: drm: msm: add explicit firmware for a702

update-initramfs uses this information to add the correct firmware to initramfs.
Needed for early kms and plymouth to work properly
---
 drivers/gpu/drm/msm/adreno/a6xx_catalog.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/msm/adreno/a6xx_catalog.c b/drivers/gpu/drm/msm/adreno/a6xx_catalog.c
index 3e6f409d1..cc4ec4ee0 100644
--- a/drivers/gpu/drm/msm/adreno/a6xx_catalog.c
+++ b/drivers/gpu/drm/msm/adreno/a6xx_catalog.c
@@ -2504,3 +2504,6 @@ static inline __always_unused void __build_asserts(void)
 	BUILD_BUG_ON(a730_protect.count > a730_protect.count_max);
 	BUILD_BUG_ON(a810_protect.count > a810_protect.count_max);
 }
+
+MODULE_FIRMWARE("qcom/a702_sqe.fw");
+MODULE_FIRMWARE("qcom/qcm2290/a702_zap.mbn");
